blakcomp/function.c

```diff
@@ -36,6 +36,7 @@ function_type Functions[] = {
 {"Nth",			NTH,		 AEXPRESSION,	AEXPRESSION,	ANONE},
 {"SetFirst",            SETFIRST,        AEXPRESSION,   AEXPRESSION,    ANONE},
 {"SetNth",              SETNTH,          AEXPRESSION,   AEXPRESSION,    AEXPRESSION, ANONE},
+{"SwapListElem", SWAPLISTELEM, AEXPRESSION, AEXPRESSION, AEXPRESSION, ANONE},
```

Comment:
I think I have messed this up and I need to adjust the server side function to be insert instead of swap. More info below.

On the client side list.c, the item the user drags (item1 or payload) is inserted at the position of item2 (target). So for example if item1 (payload) is in position 3 and item2 (target) is in position 7, then item1 will be removed from the list, item2 will drop to position 6, and then item1 will be inserted at item2's original position of 7. This is all what we're looking for.

The server receives the request to move the items, so the client inventory and the server inventory will match, but on the server side I am using a swap function. This is problematic.

In short, I need to take a look at it. And it probably makes sense to make the server side function an insert function, because otherwise the algorithms aren't the same even if they're targeting the same items. This will probably result in unpredictable behavior when the user logs out and in, and sees their inventory reordered to something they didn't expect. I have tested it and sometimes that happens but it's unclear why (lag, packet not received, or the sort functions not matching).

Reply:
Implemented new sort function MoveListElem from other PR #484